docs/docs/07-faq.md

```diff
@@ -0,0 +1,187 @@
+
+### Can I make `ExternalEvent`s idempotent?
+
+Yes, just pass the `guid` parameter in the `PutExternalEventRequest` when making the `rpc PutExternalEvent`.
+
+## Workflow Threading
+
+### Can I run tasks in parallel?
+
+To run tasks in parallel within a single `WfRun`, you need to [create a Child Thread](./05-developer-guide/08-wfspec-development/07-child-threads.md).
+
+### Can different threads share variables?
+
+A child `ThreadRun` can read and write the variables of its parent. A parent cannot view the variables of its child.
+
+### What happens when a child fails?
+
+When a child fails, the failure propagates to the parent either at:
+* The `WaitForThreadsNode`, or
+* The `ExitNode`.
+
+See the [Child Thread Docs](./05-developer-guide/08-wfspec-development/07-child-threads.md) for info on how to handle these failures.
+
+### What happens to the child when a parent fails?
+
+When a parent `ThreadRun` fails (`ERROR` or `EXCEPTION`) or moves to `HALTED`, the child will move to `HALTED`.
+
+## User Tasks
+
+### What are User Tasks?
+
+[User Tasks](./04-concepts/05-user-tasks.md) allow a workflow to wait for input from a human.
+
+### How do I add a user task to my workflow?
+
+Check out the [User Tasks](./05-developer-guide/08-wfspec-development/08-user-tasks.md) page of our `WfSpec` docs.
+
+### Do User Tasks support users and user groups?
+
+Yes, user tasks can be assigned to either a user_id or a user_group. Note that user and group identities are not managed by the LH Server.
+
+### How do I complete User Tasks?
+
+For testing purposes, you can do it [using `lhctl`](./05-developer-guide/03-lhctl.md#usertaskrun). For production applications, check out our docs on [managing User Tasks with grpc](./05-developer-guide/09-grpc/20-user-tasks.md).
+
+### How do I create a `UserTaskDef`?
+
+Check out our [Managing Metadata docs](./05-developer-guide/09-grpc/05-managing-metadata.md#usertaskdef).
+
+## Exception Handling
+
+### What is the difference between `ERROR` and `EXCEPTION`?
+
+As per our [Failure Handling Concepts Docs](./04-concepts/01-workflows.md#failure-handling), an `ERROR` represents a technical failure such as a type error, network outage, timeout, or unexpected exception thrown by a task worker. An `EXCEPTION` represents something going wrong _at the business process level_, such as an invalid credit card or an item being out of stock.
+
+### How do I catch a failure in a workflow?
+
+Check out our [Failure Handling `WfSpec` docs](./05-developer-guide/08-wfspec-development/06-exception-handling.md).
+
+### How do I throw an `EXCEPTION`?
+
+You can [throw an `EXCEPTION` from a Task Worker](./05-developer-guide/05-task-worker-development.md#throwing-workflow-exceptions). You can also make a `ThreadRun` fail with the `WorkflowThread.fail()` method.
+
+## Reliability
+
+### Is LittleHorse Fault-Tolerant?
+
+Yes, it was built into the [core DNA of our system](./02-architecture-and-guarantees.md).
```

Comment:
Every component of the LittleHorse tech stack is highly available and fault tolerant.  At the workflow layer LittleHorse provides the primitives to handle faults in a way to suit the business needs.